File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;
class Solution {
    public int[] solution(int[] answers) {
        int[] one = {1,2,3,4,5};
        int[] two = {2,1,2,3,2,4,2,5};
        int[] three= {3,3,1,1,2,2,4,4,5,5};
        int[] points = new int[3];
        for(int i = 0; i < 3; i++){
            int maxPoint = 0;
            for(int j = 0; j < answers.length; j++){
                
                if(i == 0){
                    if(answers[j] == one[j % 5]){
                        maxPoint++;
                    }
                }else if(i == 1){
                    if(answers[j] == two[j % 8]){
                        maxPoint++;
                    }
                }else if(i == 2){
                    if(answers[j] == three[j % 10]){
                        maxPoint++;
                    }
                }
            }points[i] = maxPoint;
        }
        
        
        int maxGrade = 0;
        for(int i = 0; i < 3; i++){
            maxGrade = Math.max(maxGrade, points[i]);
        }
        List<Integer> answer = new ArrayList<>();
        for(int i = 0; i < 3; i++){
            if(points[i] == maxGrade){
                answer.add(i+1);
            }
        }
        int[] result = answer.stream().mapToInt(Integer::intValue).toArray();
        return result;
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    result = []
    one = [1,2,3,4,5]
    two = [2,1,2,3,2,4,2,5]
    three = [3,3,1,1,2,2,4,4,5,5]
    one_result = 0
    two_result = 0
    three_result = 0
    for i in range(len(answers)):
        answer = answers[i]
        i_one = i % len(one)
        i_two = i % len(two)
        i_three = i % len(three)
        
        if answer == one[i_one]:
            one_result += 1
        if answer == two[i_two]:
            two_result += 1
        if answer == three[i_three]:
            three_result += 1
    
    
    high_score = max(one_result, two_result, three_result)
    if one_result == high_score:
        result.append(1)
    if two_result == high_score:
        result.append(2)
    if three_result == high_score:
        result.append(3)
    
    
    
    return result